src/View/Widget/MultiCheckboxWidget.php

```diff
@@ -113,10 +115,26 @@ public function render(array $data, ContextInterface $context)
             'idPrefix' => null,
             'templateVars' => []
         ];
-        $out = [];
         $this->_idPrefix = $data['idPrefix'];
         $this->_clearIds();
+        return implode('', $this->_renderInputs($data, $context));
+    }
+
+    protected function _renderInputs($data, $context)
+    {
+        $out = [];
         foreach ($data['options'] as $key => $val) {
+            // Grouped inputs in a fieldset.
+            if (is_string($key) && is_array($val) && !isset($val['text'], $val['value'])) {
+                $inputs = $this->_renderInputs(['options' => $val] + $data, $context);
+                $legend = $this->_templates->format('legend', ['text' => $key]);
+                $out[] = $this->_templates->format('fieldset', [
```

Comment:
Would it be reasonable to use separate templates for text and container? While one may want to change the templates for multi-checkboxes, having this affect fieldsets and legends in general may be unwelcome.

Another option might be to set a (configurable) class for the fieldset, so that they can be easily targeted via CSS.

An additional template around the fieldsets so that everything is contained in a single element might be useful too.